doublyLinkedLists.js

```diff
@@ -0,0 +1,499 @@
+        
+        // crawl until position reached
+        while (index > position) {
+
+            next = current;
+            current = current.previous;
+            previous = current.previous;
+            index--;
+
+        }
+        
+        // delete current node
+        next.previous = previous;
+        previous.next = next;
+
+        this._size--;
+
+        return current;
+        
+    },
+    
+    // insert node at specified position
+    deleteAt: function (position) {
+        
+        // if position is within bounds of list
+        if (this._size > 0 && position >= 1 && position <= this._size) {
+            
+            // find middle of list to determine shortest way to delete node at position
+            var middle = Math.ceil(this._size / 2);
+            
+            // if position is in first half of list
+            if (position <= middle) {
+                
+                // if position is 1, set head equal to newNode
+                if (position === 1) {
+                    return this.deleteFirst();
+                }
+                
+                // delete node by crawling from front
+                return this.deleteFromFront(position);
+                
+            } else {
+                
+                // if position is tail, set tail equal to newNode
+                if (position === this._size) {
+                    return this.deleteLast();
+                }
+                
+                // delete node by crawling from end
+                return this.deleteFromEnd(position);
+                
+            }
+            
+        }
+        
+        throw new Error('Position out of bounds');
+        
+    },
+    
+    // delete first node found containing the specified data
+    delete: function (data) {
+        
+        var current = this.head,
+            next,
+            previous;
+        
+        if (this._size > 0) {
+        
+            // if head contains data, delete head
+            if (data === current.data) {
+
+                this.head = current.next;
+                this._size--;
+
+                return true;
+
+            } else {
+
+                // crawl thorugh list until data is found and delete the node cointaining it
+                for (var i = 1; i < this._size - 1; i++) {
+
+                    previous = current;
+                    current = current.next;
+                    next = current.next;
+
+                    if (data === current.data) {
+
+                        previous.next = next;
+                        next.previous = previous;
+                        this._size--;
+
+                        return true;
+                    }
+                }
+            }
+        }
+        
+        // if node isn't throw an error
+        throw new Error('The specified node was not found');
+        
+    },
+    
+    // logs the full list from head to tail
+    displayForward: function () {
```

Comment:
return a string, don't log as part of your data structure that's creating useless dependencies, the consumer of this method can decide what to do with the string representation of your object, here you are forcing his hand and forcing console to be defined in your environment